src/toolkits/drawing_classifier/dc_data_iterator.cpp

```diff
@@ -40,44 +40,52 @@ void add_drawing_pixel_data_to_batch(float* next_drawing_pointer,
 
 simple_data_iterator::target_properties
 simple_data_iterator::compute_properties(
-    const gl_sarray& targets, std::vector<std::string> expected_class_labels) {
-  target_properties result;
+    const gl_sframe& data, std::string target_column_name,
+    std::vector<std::string> expected_class_labels) {
 
-  // Determine the list of unique class labels,
-  gl_sarray classes = targets.unique().sort();
+  target_properties result;
 
-  if (expected_class_labels.empty()) {
-    // Infer the class-to-index map from the observed labels.
-    result.classes.reserve(classes.size());
-    int i = 0;
-    for (const flexible_type& label : classes.range_iterator()) {
-      result.classes.push_back(label);
-      result.class_to_index_map[label] = i++;
-    }
-  } else {
-    // Construct the class-to-index map from the expected labels.
-    result.classes = std::move(expected_class_labels);
-    int i = 0;
-    for (const std::string& label : result.classes) {
-      result.class_to_index_map[label] = i++;
-    }
+  if (data.contains_column(target_column_name)) {
```

Comment:
Instead of indenting essentially the entire body of this function inside this `if` statement, could we instead have an early return?